### ISSUE TITLE
Email and Phone fields to shipping address

### ISSUE THREAD

**Issue by joosev** [Pakettikauppa/woo-pakettikauppa#123]
Currently Woocommerce Shipping object in the order does not include email nor phone number.

Is it possible to add those fields to checkout (similar to product information is added)?

**Comment by k1sul1** [comment]
![image](https://user-images.githubusercontent.com/2719615/73471047-74f8c200-4391-11ea-9850-c07d39e085fe.png)

Both are required by default. The data is in the order, and it's also set to the recipient in create_shipment_from_order:

```
$receiver->setEmail($order->get_billing_email());
$receiver->setPhone($order->get_billing_phone());
```

So it should be there?

**Comment by joosev** [comment]
That is in billing not in shipping object.

**Comment by joosev** [comment]
Sometime people order stuff as a gift so billing email and phone differs from shipping.

**Comment by k1sul1** [comment]
Ok. Is it right to assume that you need this data for pickup point notifications?  How do I send the data to Pakettikauppa API, do I add it under $info in create_shipment_from_order or what?

It's [pretty easy to add](https://gist.github.com/bavington/f325d5bee9325001bba3) fields to checkout, but that will not work if Klarna Checkout is used. I'd suggest that we add those fields to pickup_point_field_html instead.

I can add this to #124 later today, and release next week?

**Comment by joosev** [comment]
if shipping email-address exists, use that instead of billing
if shipping phone-number exists, use that instead of billing

This feature should be configurable from settings. Not all want to use this feature I think.

By default: use this feature

**Comment by joosev** [comment]
I would not put these to pickup_point_field_html because these values are needed for shipments that are not for pickup points - like for home delivery. Also the logistics companies need to communicate with the parcel receiver.

**Comment by k1sul1** [comment]
It doesn't matter what it's for, my original point about Klarna stands, extra fields added to checkout won't work. 

There is no "shipping email" or phone. WC_Order has get_billing_phone and get_billing_email methods, get_shipping_email doesn't work as it doesn't exist. 

I don't see much point in doing it as a configurable option either. You'd have to hate your customers to do that? 

> How do I send the data to Pakettikauppa API, do I add it under $info in create_shipment_from_order or what?

I still don't know what I'm supposed to do with the data even if I added the fields for it.

**Comment by joosev** [comment]
> It doesn't matter what it's for, my original point about Klarna stands, extra fields added to checkout won't work.

With Klarna, you can't order something as a gift (trying to prevent frauds).

> There is no "shipping email" or phone. WC_Order has get_billing_phone and get_billing_email methods, get_shipping_email doesn't work as it doesn't exist.

Yes. I know. I said it in the issue description. "Currently Woocommerce Shipping object in the order does not include email nor phone number."

There is no shipping email or shipping phone in Woocommerce. WC_Order does not have get_shipping_phone or get_shipping_email functions. Only get_billing_phone and get_billing_email.

That is why we need to add those fields to checkout. It does not need to extend WC_Order to provide get_shipping_email but only functionality that is similar.

Plugin should not use Woocomerce get shipping email/phone because it does not exists. We have to create our own extra fields (similar what we have done with product to add customs codes) so we can get that infromation.

> I don't see much point in doing it as a configurable option either. You'd have to hate your customers to do that?

No?

Merchants want different things. There might be merchants who don't want to have shipping email/address to be asked for any reason.

> > How do I send the data to Pakettikauppa API, do I add it under $info in create_shipment_from_order or what?
> I still don't know what I'm supposed to do with the data even if I added the fields for it.

In core/class-shipment.php:

Lines: 472-473:

      $receiver->setEmail($order->get_billing_email());
      $receiver->setPhone($order->get_billing_phone());

To be changed something like this:

```
if ($shipping_email is empty) {
      $receiver->setEmail($order->get_billing_email());
} else {
      $receiver->setEmail($shipping_email);
}

if ($shipping_phone is empty) {
      $receiver->setPhone($order->get_billing_phone());
} else {
      $receiver->setPhone($shipping_phone);
}
```

Above code is not PHP nor any other programming language - it's so called pseudo code and is used to just show the logic.

**Comment by k1sul1** [comment]
> With Klarna, you can't order something as a gift (trying to prevent frauds).

Fair enough, I did not know that. 

> Yes. I know. I said it in the issue description.

The original description was ambiguous. I wasn't sure if it's a bug or if new fields had to be added, and just pointed out that those fields indeed do not exist yet, and that there shouldn't be a bug either.

> I don't see much point in doing it as a configurable option either. You'd have to hate your customers to do that?
> No?
> Merchants want different things. There might be merchants who don't want to have shipping email/address to be asked for any reason.

The point that I was trying to make was that I don't think that making it a configurable option adds any value. It complicates things. One or two configurable options never hurt anyone but the cyclomatic complexity grows dramatically as you add more and more options. Also, [YAGNI](https://martinfowler.com/bliki/Yagni.html).

The "ship to another address" fields (that these new fields would go under?) are already hidden by default, user has to tick a box to show them. As an user, I'd expect to see mirrored fields from the billing object anyway, and if I don't fill them, the fields in billing are used, like your pseudo example describes. I may not have a degree in CS but I'm not an idiot, I know what pseudocode is.

Nevertheless, I'll implement it by the end of the month, and I'll make it configurable unless I managed to change your mind about it.
